sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -316,10 +348,95 @@ private Message createManagementMessage(String operation, String linkName) {
+        List<Message> messagesToSchedule = new ArrayList<>();
+        messagesToSchedule.add(messageSerializer.serialize(messageToSchedule));
+        Collection<HashMap<String, Object>> messageList = new LinkedList<>();
+        for (Message message : messagesToSchedule) {
+            final int payloadSize = messageSerializer.getSize(message);
+            final int allocationSize =
+                Math.min(payloadSize + MAX_MESSAGING_AMQP_HEADER_SIZE_BYTES, maxMessageSize);
+            final byte[] bytes = new byte[allocationSize];
+
+            int encodedSize;
+            try {
+                encodedSize = message.encode(bytes, 0, allocationSize);
+            } catch (BufferOverflowException exception) {
+                final String errorMessage =
+                    String.format(Locale.US,
+                        "Error sending. Size of the payload exceeded maximum message size: %s kb",
+                        maxMessageSize / 1024);
+                throw logger.logExceptionAsWarning(new AmqpException(false,
+                    AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, errorMessage, exception, getErrorContext()));
+            }
+
+            HashMap<String, Object> messageEntry = new HashMap<>();
+
+            messageEntry.put(MESSAGE, new Binary(bytes, 0, encodedSize));
+            messageEntry.put(MESSAGE_ID, message.getMessageId());
+            messageList.add(messageEntry);
+        }
+        requestBodyMap.put(MESSAGES, messageList);
+
+        return requestBodyMap;
+    }
+
     private AmqpErrorContext getErrorContext() {
         return new SessionErrorContext(fullyQualifiedNamespace, entityPath);
     }
 
+    private Mono<Void> cancelSchedule(Long[] cancelScheduleNumbers) {
+        return  isAuthorized(CANCEL_SCHEDULED_MESSAGE_OPERATION).thenMany(createRequestResponse.flatMap(channel -> {
+
+            Message requestMessage = createManagementMessage(CANCEL_SCHEDULED_MESSAGE_OPERATION,
+                channel.getReceiveLinkName());
+
+            requestMessage.setBody(new AmqpValue(Collections.singletonMap(SEQUENCE_NUMBERS, cancelScheduleNumbers)));
+            return channel.sendWithAck(requestMessage);
+        }).map(responseMessage -> {
+            int statusCode = RequestResponseUtils.getResponseStatusCode(responseMessage);
+
+            if (statusCode ==  AmqpResponseCode.OK.getValue()) {
+                return Mono.empty();
+            }
+            return Mono.error(new AmqpException(false, "Could not cancel schedule message with sequence "
+                + Arrays.toString(cancelScheduleNumbers), getErrorContext()));
+
+        })).then();
+    }
+
+    private Flux<Long> scheduleMessage(ServiceBusMessage messageToSchedule, Instant scheduledEnqueueTime) {
+
+        messageToSchedule.setScheduledEnqueueTime(scheduledEnqueueTime);
+
+        return  isAuthorized(SCHEDULE_MESSAGE_OPERATION).thenMany(createRequestResponse.flatMap(channel -> {
```

Comment:
You may need to reformat your code. There are double spaces where there shouldn't be.